pkg/trace/api/telemetry.go

```diff
@@ -0,0 +1,136 @@
+	}
+	if len(endpoints) == 0 {
+		return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+			log.Errorf("Telemetry proxy couldn't forward a request - no valid endpoints found")
+			msg := fmt.Sprintf("Telemetry proxy disabled, check configured endpoints")
+			http.Error(w, msg, http.StatusMethodNotAllowed)
+		})
+	}
+
+	transport := telemetryMultiTransport{
+		Transport: r.conf.NewHTTPTransport(),
+		Endpoints: endpoints,
+	}
+	limitedLogger := logutil.NewThrottled(5, 10*time.Second) // limit to 5 messages every 10 seconds
+	logger := stdlog.New(limitedLogger, "telemetry.Proxy: ", 0)
+	director := func(req *http.Request) {
+		req.Header.Set("Via", fmt.Sprintf("trace-agent %s", info.Version))
+		if _, ok := req.Header["User-Agent"]; !ok {
+			// explicitly disable User-Agent so it's not set to the default value
+			// that net/http gives it: Go-http-client/1.1
+			// See https://codereview.appspot.com/7532043
+			req.Header.Set("User-Agent", "")
+		}
+		req.Header.Set("DD-Agent-Hostname", r.conf.Hostname)
+		req.Header.Set("DD-Agent-Env", r.conf.DefaultEnv)
+	}
+	return &httputil.ReverseProxy{
+		Director:  director,
+		ErrorLog:  logger,
+		Transport: &transport,
+	}
+}
+
+// RoundTrip sends request first to Endpoint[0], then sends a copy of main request to every configurged
+// additional endpoint.
+//
+// All requests will be sent irregardless of any errors
+// If any request fails, the error will be logged. Only main target's
+// error will be propagated via return value
+func (m *telemetryMultiTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	if len(m.Endpoints) == 1 {
+		return m.roundTrip(req, m.Endpoints[0])
+	}
+	slurp, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		return nil, err
+	}
+	newreq := req.Clone(req.Context())
+	newreq.Body = ioutil.NopCloser(bytes.NewReader(slurp))
+	// despite the number of endpoints, we always return the response of the first
+	rresp, rerr := m.roundTrip(newreq, m.Endpoints[0])
+	for _, endpoint := range m.Endpoints[1:] {
+		newreq := req.Clone(req.Context())
+		newreq.Body = ioutil.NopCloser(bytes.NewReader(slurp))
+		if resp, err := m.roundTrip(newreq, endpoint); err == nil {
+			// we discard responses for all subsequent requests
+			io.Copy(ioutil.Discard, resp.Body)
+			resp.Body.Close()
+		} else {
+			log.Error(err)
+		}
+	}
+	return rresp, rerr
+}
+
+func (m *telemetryMultiTransport) roundTrip(req *http.Request, endpoint *config.Endpoint) (*http.Response, error) {
+	tags := []string{
+		fmt.Sprintf("endpoint:%s", endpoint.Host),
+	}
+	defer func(now time.Time) {
+		metrics.Timing("datadog.trace_agent.telemetry_proxy.roundtrip_ms", time.Since(now), tags, 1)
+	}(time.Now())
+
+	req.Host = endpoint.Host
+	req.URL.Host = endpoint.Host
+	req.URL.Scheme = "https"
+	req.Header.Set("DD-API-KEY", endpoint.APIKey)
+
+	resp, err := m.Transport.RoundTrip(req)
+	if err != nil {
+		metrics.Count("datadog.trace_agent.telemetry_proxy.roundtrip.errors", 1, tags, 1)
```

Comment:
```suggestion
		metrics.Count("datadog.trace_agent.telemetry_proxy.error", 1, tags, 1)
```
Or `roundtrip_error`. We should minimise nesting.